File: mergeSort.py
# Write a python program to perform merge sort
def mergeSort(list):
    if len(list) > 1:
        m = len(list) // 2
        L = list[:m]
        R = list[m:]
        mergeSort(L)
        mergeSort(R)
        i = j = k = 0
        while i < len(L) and j < len(R):
            if L[i] < R[j]:
                list[k] = L[i]
                i += 1
            else:
                list[k] = R[j]
                j += 1
            k += 1
        while i < len(L):
            list[k] = L[i]
            i += 1
            k += 1
        while j < len(R):
            list[k] = R[j]
            j += 1
            k += 1

list = []
n = int(input("Enter the no. of elements: "))
print("Enter the elements:",end="\n");
for i in range(n):
    val = int(input())
    list.append(val)
mergeSort(list)
print(list)

File: MergeSort.java
import java.util.*;
public class MergeSort {
    static void merge(int[] arr, int l, int m, int r) {
        int[] temp = new int[r - l + 1];
        int p1 = l, p2 = m + 1, count = 0;
        while(p1 <= m && p2 <= r) {
            if(arr[p1] < arr[p2])
                temp[count++] = arr[p1++];
            else
                temp[count++] = arr[p2++];
        }
        while(p1 <= m) 
            temp[count++] = arr[p1++];
        while(p2 <= r) 
            temp[count++] = arr[p2++];
        for(int i = r; i >= l; i--) 
            arr[i] = temp[--count];
    }

    static void mergeSort(int[] arr, int l, int r) {
        if(l < r) {
            int m = l + (r - l) / 2;
            mergeSort(arr, l, m);
            mergeSort(arr, m+1, r);
            merge(arr, l, m, r);
        }
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter the total no. of elements: ");
        int n = sc.nextInt();
        int[] arr = new int[n];
        System.out.print("Enter the elements:\n");
        for(int i = 0; i < n; i++) 
            arr[i] = sc.nextInt();
        mergeSort(arr, 0, n-1);
        for(int i = 0; i < n; i++) {
            System.out.print(arr[i] + " ");
        }
        sc.close();
    }
}
